ext/include/opentelemetry/ext/zpages/tracez_data_aggregator.h

```diff
@@ -0,0 +1,127 @@
+  
+  /**
+   * span_count_per_latency_bucket_ is a vector that stores the count of spans for each of the 9 latency buckets.
+   */
+  std::vector<int> span_count_per_latency_bucket_;
+  
+  AggregatedInformation()
+  {
+    running_spans_ = 0;
+    error_spans_ = 0;
+    latency_sample_spans_.resize(kLatencyBoundaries.size());
+    span_count_per_latency_bucket_.resize(kLatencyBoundaries.size(),0);
+  } 
+};
+
+/**
+ * This class is responsible for collecting raw data and converting it to useful information that can be made available to the user.
+ */
+class TracezDataAggregator
+{
+public:
+  TracezDataAggregator(std::shared_ptr<TracezSpanProcessor> spanProcessor);
+  
+  /** 
+   * AggregateSpans is the function that is called to update the aggregated data with newly
+   * completed and running span information
+   */
+  void AggregateSpans();
+  
+  /** 
+   * GetAggregatedData gets the aggregated span information
+   * @returns a map with the span name as key and the aggregated information for the span name as value.
+   */
+  const std::map<std::string, std::unique_ptr<AggregatedInformation>>& GetAggregatedData();
+  
+  
+private:
+  /**Instance of the span processor to collect raw data**/
+  std::shared_ptr<TracezSpanProcessor> tracez_span_processor_;
+  
+  /**
+   * Tree map with key being the name of the span and value being a vector of size 9(number of latency boundaries), 
+   * with each index value representing number of spans with a particular name and latency boundary. 
+   * A hash map is not used because the span names are displayed in sorted order
+   */
+  std::map<std::string, std::unique_ptr<AggregatedInformation>> aggregated_data_;
+  
+  /**
+   * GetLatencyBoundary returns the latency boundary to which the latency of the given recordable belongs to
+   * @ param recordable is the recordable for which the latency boundary is to be found
+   * @ returns LatencyBoundaryName is the name of the latency boundary that the recordable belongs to
+   */
+  LatencyBoundaryName GetLatencyBoundary(opentelemetry::sdk::trace::Recordable* recordable);
+  
+  /** 
+   * AggregateCompletedSpans is the function that is called to update the aggregated data of newly completed spans
+   */
+  void AggregateCompletedSpans();
+  
+  /**
+   * AggregateRunningSpans aggregates the data for running spans.
+   */
+  void AggregateRunningSpans();
+  
+  
+  /** 
+   * AggregateStatusOKSpans is the function called to update the data of spans with status code OK.
+   * @param ok_span is the span who's data is to be collected
+   */
+  void AggregateStatusOKSpans(std::unique_ptr<opentelemetry::sdk::trace::Recordable>& ok_span);
+  
+  /** 
+   * AggregateStatusErrorSpans is the function that is called to collect the information of error spans
+   * @param error_span is the error span who's data is to be collected
+   */
+  void AggregateStatusErrorSpans(std::unique_ptr<opentelemetry::sdk::trace::Recordable>& error_span);
```

Comment:
This and ok spans seems like they should be singular, since only a single span is aggregated. Another name for these functions could be UpdateStatusErrorAggregations or UpdateStatusErrorSpanAggregations

Reply:
Changed it to AggregateStatusErrorSpan